app/v2/controllers/V2MovementsController.scala

```diff
@@ -434,28 +434,59 @@ class V2MovementsControllerImpl @Inject() (
     (authActionNewEnrolmentOnly andThen acceptHeaderActionProvider(jsonOnlyAcceptHeader)).async {
       implicit request =>
         implicit val hc: HeaderCarrier = HeaderCarrierConverter.fromRequest(request)
-        persistenceService
-          .getMovements(request.eoriNumber, movementType, updatedSince, movementEORI, movementReferenceNumber, page, count, receivedUntil, localReferenceNumber)
-          .asPresentation
-          .fold(
-            presentationError => Status(presentationError.code.statusCode)(Json.toJson(presentationError)),
-            response =>
-              Ok(
-                Json.toJson(
-                  HateoasMovementIdsResponse(
-                    response,
-                    movementType,
-                    updatedSince,
-                    movementEORI,
-                    movementReferenceNumber,
-                    page,
-                    count,
-                    receivedUntil,
-                    localReferenceNumber
-                  )
+        (page, count) match {
```

Comment:
We'd need to do this for the `getMessages` endpoint as well, this solution isn't very reusable.

Instead, why don't we make a method:

```scala
def ensurePositive(parameter: Option[Int], keyName: String): EitherT[Future, PresentationError, Unit] = 
    parameter match {
        case Some(x) if x <= 0 => EitherT.leftT(BadRequest(...))
        case _ => EitherT.rightT((): Unit)
    }
``` 

and then we can just use this function in the for comprehension:

```scala
for {
    _ <- ensurePositive(page, "page")
    _ <- ensurePositive(page, "count")
   ....
} yield ....
```

and that way everything is reusable?

Reply:
Okay I'll do that